src/openapi-generator.ts

```diff
@@ -70,22 +59,14 @@ export class OpenAPIGenerator {
   private paramRefs: Record<string, ParameterObject> = {};
   private pathRefs: Record<string, Record<string, PathObject>> = {};
 
-  constructor(
-    private definitions: OpenAPIDefinitions[],
-    private config?: OpenAPIObjectConfig
-  ) {}
-
-  generateDocs(): OpenAPIObject {
-    if (!this.config) {
-      throw new Error(
-        'No config was provided when creating the OpenAPIGenerator'
-      );
-    }
+  // TODO: The order is very important here!
```

Comment:
Yes - the schemas and parameters should be generated first so that we can use ref instead of inline schemas. I opted for a sort before generating them in the other PR